src/main/webapp/go.js

```diff
@@ -150,30 +185,39 @@ function toggleHintButton(hide) { //eslint-disable-line
  * Disable lint check because getHint() is called from go.html.
  */
 function getHint() { //eslint-disable-line
-  if (destIndex >= 0) {
-    const hintArr = huntArr[destIndex].hints;
+  if (hunt.getDestIndex() >= 0) {
```

Comment:
I'm thinking this logic could live inside the ScavengerHuntManager class.  You could call hunt.getNextHint() to get the next hint, if available and manage all the hintIndex logic.  The code here would then be much simpler and only in charge of updating the DOM:

```
function getHint() {
  const nextHint = hunt.getNextHint();
  if (nextHint != "") {
    updateMessage(HINT_DISPLAY, nextHint);
  }
}
```